engine/src/test/java/io/zeebe/engine/processor/workflow/incident/JobActivationIncidentTest.java

```diff
@@ -0,0 +1,130 @@
+  private static final String LARGE_TEXT = "x".repeat((int) (MAX_MESSAGE_SIZE / 4));
+  private static final String JOB_TYPE = "test";
+  private static final BpmnModelInstance WORKFLOW =
+      Bpmn.createExecutableProcess("process")
+          .startEvent()
+          .serviceTask("task", t -> t.zeebeJobType(JOB_TYPE))
+          .endEvent()
+          .done();
+
+  private static long workflowKey;
+
+  @Rule
+  public RecordingExporterTestWatcher recordingExporterTestWatcher =
+      new RecordingExporterTestWatcher();
+
+  @BeforeClass
+  public static void init() {
+    workflowKey =
+        ENGINE
+            .deployment()
+            .withXmlResource(WORKFLOW)
+            .deploy()
+            .getValue()
+            .getDeployedWorkflows()
+            .get(0)
+            .getWorkflowKey();
+  }
+
+  @Test
+  public void shouldRaiseIncidentWhenActivatingJobThatIsTooBigForMessageSize() {
+    // given
+    final var workflowInstanceKey = ENGINE.workflowInstance().ofBpmnProcessId("process").create();
+
+    for (int i = 0; i < 4; i++) {
+      ENGINE
+          .variables()
+          .ofScope(workflowInstanceKey)
+          .withDocument(Map.of(String.valueOf(i), LARGE_TEXT))
+          .update();
+    }
+
+    // when
+    final var activationResult =
+        ENGINE.jobs().withMaxJobsToActivate(1).withType(JOB_TYPE).byWorker("dummy").activate();
+
+    // then
+    Assertions.assertThat(activationResult.getValue().getJobs()).isEmpty();
+    Assertions.assertThat(activationResult.getValue().isTruncated()).isTrue();
+
+    final var incidentCommand =
+        RecordingExporter.incidentRecords()
+            .withIntent(IncidentIntent.CREATE)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    assertThat(incidentCommand.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR)
+        .hasBpmnProcessId("process")
+        .hasWorkflowKey(workflowKey)
+        .hasWorkflowInstanceKey(workflowInstanceKey)
+        .hasElementId("task");
+  }
+
+  @Test
+  public void shouldMakeJobActivatableAfterIncidentIsResolved() {
+    // given
+    final var workflowInstanceKey = ENGINE.workflowInstance().ofBpmnProcessId("process").create();
+
+    for (int i = 0; i < 4; i++) {
+      ENGINE
+          .variables()
+          .ofScope(workflowInstanceKey)
+          .withDocument(Map.of(String.valueOf(i), LARGE_TEXT))
+          .update();
+    }
+    ENGINE.jobs().withMaxJobsToActivate(1).withType(JOB_TYPE).byWorker("dummy").activate();
+
+    // when
+    final var incidentCommand =
+        RecordingExporter.incidentRecords()
+            .withIntent(IncidentIntent.CREATE)
+            .withWorkflowInstanceKey(workflowInstanceKey)
+            .getFirst();
+
+    ENGINE
+        .variables()
+        .ofScope(workflowInstanceKey)
+        .withDocument(
+            Map.of("0", "lorem ipsum", "1", "lorem ipsum", "2", "lorem ipsum", "3", "lorem ipsum"))
+        .update();
```

Comment:
An alternative would be to limit the fetched variables (e.g. fetch 3 instead of 4 variables). Maybe, we can create another test case for this.